jobs/utils.py

```diff
@@ -388,3 +390,339 @@ def clean_results(self, df: pd.DataFrame) -> pd.DataFrame:
+            self.base_url = 'https://api.pandascore.co/lol/matches/'
+            self.token = 'QyQjU22tZ4Ks7h26GJQYLP8JjycUujGZDewAYqpz6RNNj1aM6xo' # to assign through environment variable
+            self.leagues_panda = {
+                'LEC': 4197,
+                'LCK': 293,
+                'LPL': 294
+            }
+        pass
+
+    def identifty_team_names(self, df) -> pd.DataFrame:
+        '''
+        Fetch long labels from the database, and use them to replace short labels fetched when scraping.
+
+        parameters:
+        -----------
+        df: Pandas Dataframes
+        '''
+        if self.conn:
+            teams = pd.read_sql_query("SELECT DISTINCT short_label, long_label FROM teams", self.conn)
+        else:
+            teams = pd.DataFrame(
+                {
+                    'short_label': ['G2', 'TH', 'VIT', 'BDS', 'SK', 'RGE', 'MKOI', 'GX', 'FNC', 'KC'],
+                    'long_label': [
+                        'G2 Esport', 'Team Heretics', 'Vitality', 'Team BDS', 'SK Gaming',
+                        'Rogue', 'Movistar KOI', 'Giant X', 'Fnatic', 'Karmine Corp'
+                    ]
+                }
+            )
+        for column in ['team_1', 'team_2']:
+            df = df.merge(teams, how='left', left_on=column, right_on='short_label')
+            df[column] = df['long_label']
+            df = df.drop(['short_label', 'long_label'], axis=1)
+        return df
+
+    def insert_future_games(self, df: pd.DataFrame) -> None:
+        '''
+        Insert the new games in the database
+
+        parameters:
+        -----------
+        df: Pandas DataFrame containing cleaned and formated data.
+        '''
+        df.to_sql(name='game', con=self.conn, if_exists='append', index=False)
+
+    def insert_new_league_serie(self, df: pd.DataFrame) -> None:
+        '''
+        Look into the upcoming games and insert to the db league if it's not already there.
+
+        parameters:
+        ----------
+        df: pd.DataFrame
+            DataFrame containing the upcoming games.
+
+        return:
+        -------
+        None
+        '''
+        # Querying current league table
+        if self.conn:
+            leagues = pd.read_sql_query("SELECT * FROM league", self.conn)
+        else:
+            leagues = pd.DataFrame(
+                {
+                    'id': [1,2,3],
+                    'leaguename': ['MSI 2024', 'LEC spring 2024', 'LEC summer 2024']
+                }
+            )
+        # Fetching all available leagues
+        temp_df = df.merge(leagues, how='left', left_on='league_name', right_on='leaguename')
+        temp_df['leaguename'] = temp_df['leaguename'].fillna('to_insert')
+        insert_df = temp_df[temp_df['leaguename'] == 'insert'][['id', 'league_name']].drop_duplicates()
+        # Creating new ID based on the maximum available ID, and incremeting by 1.
+        insert_df['id'] = range(temp_df['id'].max(), temp_df['id'].max() + len(insert_df))
```

Comment:
Un id est automatiquement généré lors de l'insertion normalement, il ne faut théoriquement l'utiliser que si besoin d'update une ligne.
```suggestion
        # insert_df['id'] = range(temp_df['id'].max(), temp_df['id'].max() + len(insert_df))
```